### PR TITLE
Add Nim (from source)

### PR DESCRIPTION
**Pull request by pb-cdunn**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by pb-cdunn** [comment]
Alternative to #10045

**Comment by pb-cdunn** [comment]
@conda-forge/help-c-cpp, This is my first feedstock.

`nim` is a compiler which can generate C, C++, or Javascript from Nim source-code. It is itself a C-compiled executable. The Nim libs need to be available to users of `nim`. `nimpretty` and `nimble` are useful additional tools.

After this is merged, I will submit a follow-up PR for OSX and Win.

**Comment by pb-cdunn** [comment]
@conda-forge/staged-recipes ,

Seems to build ok. Can this be merged?

**Comment by pb-cdunn** [comment]
ping @conda-forge/staged-recipes

**Comment by pb-cdunn** [comment]
I am willing to be listed in the maintainers list.

**Comment by pb-cdunn** [comment]
@conda-forge/staged-recipes We need this in Bioconda. Can we get this merged?

**Comment by pb-cdunn** [comment]
@marcelotrevisani @mariusvniekerk @xhochy @scopatz ,

This package is much simpler than it sounds. Nim simply generates C code (or C++), and then calls the C compiler to build. It's not that different than **make** or **meson**.

Right now, to use Nim in Bioconda I have to download (and maybe build) the Nim compiler inside the recipe, which takes a few minutes. This recipe will reduce the burden on CircleCI.

(I would close #10045, but if you'd rather simply install the nightly prebuilt, that would work fine for me.)

File: recipes/nim/build.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# inject compilers
+echo "gcc.exe = \"$CC\"" >> config/nim.cfg
+echo "gcc.linkerexe = \"$CC\"" >> config/nim.cfg
+echo "clang.exe = \"$CC\"" >> config/nim.cfg
+echo "clang.linkerexe = \"$CC\"" >> config/nim.cfg
+./build.sh
+bin/nim c koch
+./koch tools
+
+ls -larth
```
Comment (xhochy) [inline recipes/nim/build.sh]: These `ls` commands shouldn't be needed.

File: recipes/nim/build.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ls -larth bin/
+ls -larth lib/
+ls -larth config/
+
+build_dir=$(pwd)
+
+mkdir -p "${PREFIX}"
+cd "${PREFIX}" || exit
```
Comment (xhochy) [inline recipes/nim/build.sh]: Why the `|| exit`?

File: recipes/nim/build.sh (hunk 1 above)
Comment (scopatz) [reply recipes/nim/build.sh]: I think they are OK for debugging though

File: recipes/nim/build.sh (hunk 2 above)
Comment (scopatz) [reply recipes/nim/build.sh]: Yeah, it should be `exit 1`, but conda-build passes in `set -e` so this is not strictly needed

File: recipes/nim/build.sh (hunk 1 above)
Comment (pb-cdunn) [reply recipes/nim/build.sh]: Good. I will delete them in a future PR. I plan to add OSX and Win support later anyway.

File: recipes/nim/build.sh (hunk 2 above)
Comment (pb-cdunn) [reply recipes/nim/build.sh]: Copy/paste error from elsewhere. Deleted.

**Comment by pb-cdunn** [comment]
Shall I submit updates as PRs against the new **nim-feedstock**?

**Comment by xhochy** [comment]
> Shall I submit updates as PRs against the new **nim-feedstock**?

Yes

**Pull request merged**